### PR TITLE
[RISCV][TTI] Support fdiv/udiv/sdiv/srem/urem in getArithmeticInstrCost

### PR DESCRIPTION
**Pull request by arcbbb**
This patch made following changes:
1. Support ISD FDIV/UDIV/SDIV/UREM/SREM
2. Classify instructions which cost the same

### CODE REVIEW

**Review by preames — CHANGES_REQUESTED** [review]
The code you're replacing is already linear in LMUL; your test changes are caused by adding new instructions to the switch.

Please update your review description to match and to justify the change,.

File: llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp [hunk 1]
```diff
@@ -1612,29 +1612,56 @@ InstructionCost RISCVTTIImpl::getArithmeticInstrCost(
...
+  case ISD::UDIV:
+    Op = RISCV::VDIV_VV;
+    break;
+  case ISD::SREM:
+  case ISD::UREM:
+    Op = RISCV::VREM_VV;
+    break;
   case ISD::FADD:
   case ISD::FSUB:
   case ISD::FMUL:
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp:1648]: It's weird that FMUL uses VFADD.

**Comment by topperc** [comment]
What's the history of why the cost model tests use `-riscv-v-fixed-length-vector-lmul-max=1`? That prevents fixed vector tests from covering LMULs larger than 1.

**Comment by lukel97** [comment]
> What's the history of why the cost model tests use `-riscv-v-fixed-length-vector-lmul-max=1`? That prevents fixed vector tests from covering LMULs larger than 1.

First commit adding vector cost model tests I can find is https://reviews.llvm.org/D99142, presumably back when fixed vector support was new

File: llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp [hunk 2]
```diff
@@ -1612,29 +1612,58 @@ InstructionCost RISCVTTIImpl::getArithmeticInstrCost(
   if (Op2Info.isConstant())
     ConstantMatCost += getConstantMatCost(1, Op2Info);
 
+  // Assuming instructions falling through the switch-cases have the same cost
```
[the remark this answers is not in the document]
Comment (arcbbb) [reply llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp]: Exactly! That captures what I was trying to convey.  Thank you!

**Comment by arcbbb** [comment]
ping

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**